### ISSUE TITLE
Weighted and unweighted percentages of known words

### ISSUE THREAD

**Issue by bilbe** [Hedera-Lang-Learn/hedera#146]
We want to see the percentage of all known words including duplicates as well as just unique lemmata.

**Comment by jtauber** [comment]
@joshuagetega have you started on this? Happy to work with you on it.

**Comment by joshuagetega** [comment]
Yes. @jtauber. I am working on the first part that says, "We want to see the percentage of all known words including duplicates as well as just unique lemmata." You wanna take on the second part that says, "I also want to see how many times those occur in the passage as well as the Logeion frequency". The 2 parts look like they can be worked on independently of one another.

**Comment by joshuagetega** [comment]
Hi @IvyJL, regarding seeing percentage of known words, is this referring to percentage of known words, say words with a familiarity greater than or equal to 2 out of 5, in a personal vocab list? I ask because I can see that in the Teacher view of a lemmatized text, there is a 'known words' percentage that varies depending on which vocab list (e.g. Shelmerdine 1-6 or CLC, etc) you select.

**Comment by IvyJL** [comment]
Hi, Josh,

I think there may be two questions here. First just to clarify about the weighted percentages:

Right now we show a percentage of words in the lemmatized text compared to the selected vocab list. So if puer is on the vocab list, but femina is not, a text with puer, pueri, puero, and femina will show 75% because  3/4 words are on the vocab list.

I'd like the second chart to calculate the same situation as 50% because 3 of the 4 words are all forms of the same headword (puer). So if you know all the forms of puer (and don't know femina), you know 50% of the unique headwords (puer and femina) in the text.

The second question is how does the personal word list come into all this.

At our last meeting, we talked about having the personal list be a choice in the same drop-down as the other lists. 
So when that is possible, we need to consider what specific level of knowledge counts as known. We should use the same minimum that we do in the Reader interface, which as you say is 2 out of 5.

Please, let me know if that's not clear.

Best,
Ivy

**Comment by joshuagetega** [comment]
Hi @IvyJL, thanks for the response. Sorry I did not get back sooner.

This makes sense now. I suppose I can position the second chart, i.e. the one to show 50% in the puer example, right below the first chart... I'll approach this one step at a time. Let me get that logic working and then we can figure out any aesthetics afterwards. I'll then get to the personal word list part next.

**Comment by IvyJL** [comment]
No worries @joshuagetega ! Your approach sounds very sensible.
